apm-application-toolkit/apm-toolkit-trace/src/main/java/org/apache/skywalking/apm/toolkit/trace/CorrelationContext.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.apm.toolkit.trace;
+
+import java.util.Optional;
+
+/**
+ * CorrelationContext is the interactive API for end user to put/set custom data.
+ */
+public class CorrelationContext {
```

Comment:
There are two `CorrelationContext` in your codes. Seems strange. In the old APIs, we have `TraceContext`(for users) and `ContextManager`/`TracingContext` for internal. I suggest we fix the duplicated naming issue.

Such as for users, we could have `TraceContext#putCorrelation`. @kezhenxu94 @mrproliu WDYT?

Reply:
Looks good to me

Reply:
Sure. I could change to `TraceContext#putCorrelation` and `TraceContext#getCorrelation`, replace the `CorrelationContext` toolkit. 
I think it's the same for users, they refer to the document and use them.